Remove src/backend

TueeNguyen opened the pull request:
<!--
Thanks for sending a pull request!
- if this is your first time, please ensure you have read through our contributor guide: https://github.com/Seneca-CDOT/telescope/blob/master/docs/CONTRIBUTING.md
-->

## Issue This PR Addresses
Part of #2777 
- I'm removing the code here
- Will change the docs after this PR
<!--
1. Automatically close the issue when this PR is merged
    USAGE: Fixes #<issue number>
2. If your PR addresses an issue but does not close it
    USAGE: #<issue number> <reason>(i.e. This issue was worked on by @user and myself and his PR should close the issue.)
-->

## Type of Change

<!-- bug fix, feature, documentation, UI, etc. -->

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
- Remove most of backend code
<!-- Please add a detailed description of what this PR does and why it is needed -->

## Steps to test the PR
- `cp config/env.development .env`
- `pnpm services:start parser studio db kong rest meta`
- `pnpm db:init`

- I guess if `pnpm services:start` run normally that's fine
- Need to keep in mind if other parts of telescope use anything from legacy backend
- Need to merge and test on staging to know exactly what will happen
<!-- Please add steps to build the changes in your PR locally so that peers can test your PR
    Example:
    - `npm install`
    - Go to '...'
    - Click on '....'
    - Scroll down to '....'
    - See error

    Please remember, the more clear you are, the faster your PR will be reviewed and merged.
 -->

## Checklist

<!-- Before submitting a PR, address each item -->

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

TDDR commented:
I did a `gh pr checkout 3413` and then `pnpm services:start` and got this error 
![image](https://user-images.githubusercontent.com/59970326/162100547-b456fee8-f2b4-419d-bb41-f4f59d895a52.png)

Did I do something wrong?

TueeNguyen commented:
@TDDR Ah they timed out, can you try again?

TDDR reviewed (approved):
Tested out on local and all seems to be working. I'm not sure of the greater ramifications of removing all this code but it looks to be working.

rclee91 commented on package.json:
```diff
@@ -48,46 +46,10 @@
-    "sanitize-html": "2.6.1",
-    "set-interval-async": "2.0.3",
-    "stoppable": "1.1.0"
+    "supertest": "6.1.6"
```
`supertest` is already in `devDependencies` (line 118), was the one under `dependencies` added by accident?

TueeNguyen commented on docker/development.yml:
```diff
@@ -20,8 +20,6 @@ services:
         # Web front-end URL
         - WEB_URL=${WEB_URL}
-        # Telescope 1.0 API URL
-        - API_URL=${API_URL}
```
Figure out if `API_URL` is used

JerryHue reviewed (dismissed):
I went through all files with Tue on a meeting, so I am sure that this can be approved now. We have to be aware, though, that we modified the CI workflows, so when we merge, we gotta be attentive to the build status.

humphd commented on config/env.development:
```diff
@@ -187,22 +187,6 @@ DEPENDENCY_DISCOVERY_URL=http://localhost/v1/dependency-discovery
-# PORT is the port used by the web server
-PORT=3000
-
-# API_URL is the URL of the Telescope Web API server.  Change this to
```
The only thing we lose by doing this is the ability to work on the front-end locally, but use the staging server as your back-end.  Probably that's OK, and we should migrate to using GitPod or EC2 cloud instances instead, so you can run everything at once.

humphd commented on config/env.development:
```diff
@@ -187,22 +187,6 @@ DEPENDENCY_DISCOVERY_URL=http://localhost/v1/dependency-discovery
-#   API_URL=https://telescope.cdot.systems
-API_URL=http://localhost:3000
-
 # PROXY_FRONTEND=1 will allow proxying the Next dev server (http://localhost:8000)
```
You can get rid of this next block too (`PROXY_FRONTEND=`) is only used in the backend.

humphd commented on config/env.development:
```diff
@@ -187,22 +187,6 @@ DEPENDENCY_DISCOVERY_URL=http://localhost/v1/dependency-discovery
 # NODE_ENV should be one of "development" or "production"
```
I wonder if we can remove this entire `Telescope 1.0 Legacy Environment` block?  Is anything in here needed?

TueeNguyen replied:
Parser still uses these ones 
```yml
      - NODE_ENV
      - PARSER_PORT
      - ELASTIC_MAX_RESULTS_PER_PAGE
      - ELASTIC_URL=http://elasticsearch
      - ELASTIC_PORT=9200
```

TueeNguyen replied:
Seems like only `parser` uses those variables. I can do some detective work, move in-use variables to parser section

humphd commented on docker/production.yml:
```diff
@@ -13,69 +13,7 @@ version: '3'
   ##############################################################################
-  # Telescope 1.0: Front-end and Legacy (pre-microservices) Back-end
-  ##############################################################################
-  telescope:
```
Bye-bye `telescope` container, you were good to us.

humphd commented on package.json:
```diff
@@ -13,8 +13,8 @@
     "develop": "pnpm dev",
-    "eslint": "TIMING=1 eslint --config .eslintrc.js --ignore-path .gitignore \"./src/backend/**/*.js\" \"./test/**/*.js\"",
-    "eslint-fix": "eslint --config .eslintrc.js --ignore-path .gitignore \"./src/backend/**/*.js\" \"./test/**/*.js\" --fix",
+    "eslint": "TIMING=1 eslint --ignore-path .gitignore",
```
If we're removing the `eslintrc` from the root, do we need these?  cc @menghif

menghif replied:
We can remove both `eslint` and `eslint-fix`.

humphd commented on src/web/app/src/config.ts:
```diff
@@ -24,31 +21,26 @@ const title = `Telescope`;
 const author = `SDDS Students and Professors`;
 const keywords = `Telescope, Seneca College, Open Source, Programming, Software Development, Blogs, Education, CDOT, SDDS, Students, Professors, Learning, Teaching`;
-const image = `${telescopeUrl}/logoImages/apple-splash-1334-750.jpg`;
+const image = `/logo.svg`;
```
Is an SVG compatible with a JPG in the ways we use this?

TueeNguyen replied:
So the `image` is actually being used here
https://github.com/Seneca-CDOT/telescope/blob/6046273fba4b89ce6281f7a34cafbdec7ef222dc/src/web/app/src/pages/_document.tsx#L6-L36
We tested it out by starting the services and there wasn't a problem

humphd reviewed (changes requested):
Nice work.  A few questions.

menghif commented on package.json:
```diff
@@ -13,8 +13,8 @@
-    "eslint-fix": "eslint --config .eslintrc.js --ignore-path .gitignore \"./src/backend/**/*.js\" \"./test/**/*.js\" --fix",
+    "eslint": "TIMING=1 eslint --ignore-path .gitignore",
+    "eslint-fix": "eslint --ignore-path .gitignore --fix",
     "lint": "pnpm turbo run lint && pnpm eslint",
```
Change this to: `"lint": "pnpm turbo run lint",`

humphd commented:
This needs a rebase.  Also, where are we at with the changes it makes?  Can we do this soon?

TueeNguyen commented:
I've removed some config code in `/config`. It seems like all of `Telescope 1.0 Legacy Environment` variables are used. I changed it to `Environment` and rebased

humphd reviewed (changes requested):
This looks right, but running it locally, the parser crashes with a missing service role key for Supabase:

```
{"level":50,"time":1649962783990,"pid":7,"hostname":"5b99780d3f7e","err":{"type":"Error","message":"Invalid authentication credentials","stack":"Error: Invalid authentication credentials\n    at getAllFeeds (/app/src/utils/supabase.js:21:13)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async processAllFeeds (/app/src/parser.js:62:19)"},"msg":"Error queuing feeds"}
```

Make sure you pass this into the environment for the parser container in docker-compose.yml

TueeNguyen commented:
@humphd Ah, I think it's failing because parser is using staging supabase and the `SERVICE_ROLE_KEY` is incorrect

humphd commented:
So then we can't use staging like this, since we'll never have the service role key, and depend on it.  Can you switch to use the local one?

TueeNguyen commented:
I wonder how this [PR](https://github.com/Seneca-CDOT/telescope/pull/3363) ran normally

TueeNguyen commented:
There's a problem if we use local supabase, we won't have all the feeds in the table already.

humphd commented:
Can we solve that problem in a follow-up?

TueeNguyen commented:
Sure!

TueeNguyen commented:
Changed `SUPABASE_URL` to http://kong:8000/` and rebased.

TDDR commented:
I think if you fix the merge conflicts this will be good to merge.

rclee91 commented:
Just want to note that running on the current master with `pnpm services:start` (with no backend), has no posts for me. 
I'll need time to test this branch specifically. 

Unless it might be a me problem. If someone else can test.

rclee91 commented:
Tested this PR and for me, I'm still not getting any posts with `pnpm:services start`
On `parser` it connects to `ElasticSearch` and that's it.

TueeNguyen commented:
@rclee91 did you run db migrate? The supabase database might be empty

rclee91 commented:
> @rclee91 did you run db migrate? The supabase database might be empty. Acutally, lemme update the test steps

Before I do that. In the case my potato runs out of storage, what do I run to get rid of the database

TueeNguyen commented:
@rclee91 haha don't worry, you can remove the volume in docker desktop.

TDDR commented:
When I followed your testing instructions, I got this on the second step.
![image](https://user-images.githubusercontent.com/59970326/164119087-0b515b58-d454-4127-a2f4-8eef4025d808.png)

TueeNguyen commented:
@TDDR I think you can skip that step if it's not successful

DukeManh commented:
I updated the instruction a bit, removed `prisma pull` and added `cp env.example .env.

TDDR commented:
I also updated the testing by adding another `cp env.example .env` after changing to the `tools/migrate` folder. After that I was successfully migrate and sync. Everything appears to be working.

TueeNguyen commented:
Thanks guys

TueeNguyen commented:
@humphd It's on the top under `Steps to test this PR` section

humphd reviewed (approved):
> I guess if pnpm services:start run normally that's fine

This won't work in the steps above: the parser will wait for the queue to empty before trying to re-load the feeds, which will never happen since the queue is already empty.  You have to restart the `parser` container and then it runs.  That's a bug to fix in a follow-up.

I think this is working as expected.

I want to take the steps above and automate them with a script.  It shouldn't be this hard to get the local environment to "boot."  We need an issue filed on that too.

Please file these issues and this is good to go.

humphd reviewed (approved):
If this passes CI, r+ from me.

Kevan-Y reviewed (dismissed):
Tested locally, LGTM

humphd commented:
Conflicts need to be resolved.  Let's get this finished and landed so we can test, @TueeNguyen

TueeNguyen commented:
Conflicts resolved!

humphd reviewed (approved):
If this passes CI, land it.

Kevan-Y commented on docker/docker-compose.yml:
```diff
@@ -265,9 +265,7 @@ services:
-        - WEB_URL
-        # Telescope 1.0 API URL
-        - API_URL
+        - WEB_URL=${WEB_URL}
```
I think `WEB_UR` is enough, we can just forward the args.

```suggestion
        - WEB_URL
```

TDDR commented:
Do we know why this is failing CI?

TueeNguyen commented:
@TDDR it's good now
